File: src/basics/scripts/talker.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import String

def talker():
	my_talker = rospy.Publisher('the_talker', String, queue_size=1)
	rospy.init_node('my_node', anonymous=True)
	rate = rospy.Rate(1)
	while not rospy.is_shutdown():
		the_talk = "hi, hello, bro..!"
		my_talker.publish(the_talk)
		rospy.loginfo("I just said hi, hello to my bro.")
		rate.sleep()

if __name__ == '__main__':
	try:
		talker()
	except rospy.ROSInteruptException:
		pass

File: src/basics/src/talker.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

int main(int argc, char **argv){
	
	ros::init(argc, argv, "talker");
	ros::NodeHandle my_node;
	ros::Publisher my_talker = my_node.advertise<std_msgs::String>("the_talker",1);
	ros::Rate loop_rate(2);

	while(ros::ok()){
		std_msgs::String the_talk;
		the_talk.data = "hi, hello, bro..!";
		my_talker.publish(the_talk);
		ROS_INFO("I just said hi, hello to my bro.");
		ros::spinOnce();
		loop_rate.sleep();
	}

	return 0;
}
